servicebroker/test/routes/binding_test.js

```diff
@@ -0,0 +1,204 @@
+      it("return a 400", function(done) {
+        supertest(server)
+          .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+          .set("Authorization", "Basic " + auth)
+          .send({ "app_guid": appId })
+          .expect(400)
+          .expect('Content-Type', /json/)
+          .expect({"description":messageUtil.getMessage("POLICY_REQUIRED")}, done);
+      });
+    });
+
+    context('when there is no record', function() {
+      it("creates a new binding with 201", function(done) {
+        initNockBind201();
+        supertest(server)
+          .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+          .set("Authorization", "Basic " + auth)
+          .send({ "app_guid": appId, "parameters": policy })
+          .expect(201)
+          .expect('Content-Type', /json/)
+          .expect({}, done);
+      });
+      context("when the api server return error", function() {
+        it("return a 400", function(done) {
+          initNockBind400();
+          supertest(server)
+            .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+            .set("Authorization", "Basic " + auth)
+            .send({ "app_guid": appId, "parameters": policy })
+            .expect(400)
+            .expect('Content-Type', /json/)
+            .expect({}, done);
+        });
+        it("return a 500", function(done) {
+          initNockBind500();
+          supertest(server)
+            .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+            .set("Authorization", "Basic " + auth)
+            .send({ "app_guid": appId, "parameters": policy })
+            .expect(500)
+            .expect('Content-Type', /json/)
+            .expect({}, done);
+        });
+      });
+    });
+
+    context('when an binding already exists', function() {
+
+      beforeEach(function(done) {
+        initNockBind201();
+        supertest(server)
+          .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+          .set("Authorization", "Basic " + auth)
+          .send({ "app_guid": appId, "parameters": policy })
+          .expect(201)
+          .expect('Content-Type', /json/)
+          .expect({}, done);
+      });
+
+      context('when erviceInstanceId and appId are conflict with an existed one', function() {
+        it('returns a 409', function(done) {
+          supertest(server)
+            .put("/v2/service_instances/" + serviceInstanceId + "/service_bindings/" + bindingId)
+            .set("Authorization", "Basic " + auth)
+            .set('Accept', 'application/json')
+            .send({ "app_guid": appId, "parameters": policy })
+            .expect(409)
+            .expect('Content-Type', /json/)
+            .expect({}, done);
+        });
+      });
+      context('when bindingId are conflict with an existed one', function() {
```

Comment:
There are two scenarios that will return 409. One  is an app has bound to another service instance and the other is trying to insert a binding id which already in the db.